ko/pl.rst

```diff
@@ -0,0 +1,1383 @@
+
+        -- 내부 프로시저
+        PROCEDURE inner_proc AS
+            a INT := 5;
+            b FLOAT;
+        begin
+            -- 여기서 a = 5, b는 FLOAT 타입
+        end;
+
+    BEGIN
+        -- 여기서 a = 3, b는 VARCHAR 타입
+
+        -- Block 실행문
+        DECLARE
+            a INT := 7;
+            b DATETIME;
+        BEGIN
+            -- 여기서 a = 7, b는 DATETIME 타입
+        END;
+
+        -- 다시 a = 3, b는 VARCHAR 타입
+    END;
+
+상수 선언
+=========
+::
+
+    <constant_decl> ::=
+        <identifier> CONSTANT <type_spec> [ NOT_NULL ] <value_part> ;
+
+    <value_part> ::= { := | DEFAULT } <expression>
+
+* *type_spec*: :ref:`데이터 타입 <types>` 절에서 설명한 시스템 제공 타입
+
+상수 선언에는 필수적으로 값 지정이 포함되어야 한다.
+
+.. code-block:: sql
+
+    CREATE PROCEDURE test_proc AS
+        a CONSTANT INT NOT NULL := 3;
+        b CONSTANT VARCHAR := 's';
+        c CONSTANT FLOAT;        -- 에러
+    BEGIN
+        ...
+    END;
+
+Exception 선언
+==============
+
+::
+
+    <exception_decl> ::=
+        <identifier> EXCEPTION ;
+
+사용자가 원하는 이름의 Exception을 선언할 수 있다.
+이렇게 선언된 Exception을 :ref:`RAISE <raise>` 문과 예외처리의 :ref:`WHEN <block_stmt>` 구에서 사용할 수 있다.
+
+.. code-block:: sql
+
+    CREATE FUNCTION my_func(n INT) RETURN INT AS
+        negative_argument EXCEPTION;
+        too_big_argument EXCEPTION;
+    BEGIN
+        IF n < 0 THEN
+            RAISE negative_argument;
+        ELSIF n > 100 THEN
+            RAISE too_big_argument;
+        END IF;
+        ...
+
+    EXCEPTION
+        WHEN negative_argument THEN
+            put_line('error: negative argument ' || n);
+            return -1;
+        WHEN negative_argument THEN
+            put_line('error: too big argument ' || n);
+            return -2;
+    END;
+
+.. _cursor_decl:
+
+커서 선언
+=========
+::
+
+    <cursor_decl> ::=
+        CURSOR <identifier> [ ( <seq_of_cursor_parameters> ) ] IS <select_statement> ;
+
+    <seq_of_cursor_parameters> ::= <cursor_parameter> [, <cursor_parameter>, ...]
+    <cursor_parameter> ::= <identifier> [ IN ] <type_spec>
+
+* *type_spec*: :ref:`데이터 타입 <types>` 절에서 설명한 시스템 제공 타입
+
+커서에도 프로시저/함수와 유사하게 파라메터를 선언할 수 있지만 오직 IN 파라메터만 선언할 수 있다는 차이가 있다.
+이 파라메터를 *select_statement* 문 안에서 참조할 수 있다.
+커서를 :ref:`OPEN <cursor_manipulation>` 할 때 이 파라메터에 실제 선언된 갯수와 타입이 일치하도록
+인자값을 채워 해당 SELECT 문을 실행한다.
+
+.. code-block:: sql
+
+    CREATE PROCEDURE test_proc(name VARCHAR, year INTEGER) AS
+
+        CURSOR my_cursor(a VARCHAR, y INTEGER) IS SELECT host_year, score FROM history
+            WHERE athlete = a and host_year >= y;
+
+        target_year INT;
+        target_score VARCHAR(10);
+    BEGIN
+
+        OPEN my_cursor(name, year);
+        LOOP
+            FETCH my_cursor INTO target_year, target_score;
+            EXIT WHEN my_cursor%NOTFOUND;
+            put_line('host_year: ' || target_year || ' score: ' || target_score);
+        END LOOP;
+        CLOSE my_cursor;
+    END;
+
+커서는 위 예제처럼 명시적으로 OPEN, FETCH, CLOSE 실행문을 통해 이용할 수도 있고,
+이들 동작이 암묵적으로 이루어지는 For-Loop 문을 통해서 이용할 수도 있다.
```

Comment:
We need to warn about not closing the cursor.

Reply:
It would be good to warn it. But in general, compilers does not warn on unclosed resources such as file pointers, malloced memory area, network sockets.
It is possible that JDBC automatically closes unclosed cursors when connections are closed. I'll check it.

Reply:
It meant that the manual needed a warning text.

Reply:
Yes, I'll add some warning.